misc.Rmd

```diff
@@ -2,56 +2,71 @@
+There are other package components that are either required, such as a `DESCRIPTION` file, or highly recommended, such as tests and documentation.
+
+The next few parts of the book are organized around important concepts: dependencies, testing, and documentation.
+But before we dig in to those topics, this chapter demystifies some package parts that are not needed in every package, but that are nice to be aware of.
+
+## Other directories
+
+Here are some top-level directories you might encounter in an R source package, in rough order of importance and frequency of use:
+
+-   `src/`: source and header files for compiled code, most often C and C++.
+    This is an important technique that is used to make R packages more performant and to unlock the power of external libraries for R users.
+    As of the second edition, the book no longer covers this topic, since a truly useful treatment of compiled code requires more space than we can give it here.
+    The tidyverse generally uses the [cpp11 package](https://cpp11.r-lib.org) to connect C++ to R and [Rcpp](https://www.rcpp.org) is the most well-established package for integrating R and C++.
 
 -   `inst/`: for arbitrary additional files that you want include in your package.
     This includes a few special files, like the `CITATION`, described below.
 
+-   `tools/`: auxiliary files needed during configuration, usually found in the company of a `configure` script.
+    We discuss this more below.
+
 -   `demo/`: for package demos.
-    These were useful prior to the introduction of vignettes, but are no longer recommended.
-    See below.
+    We regard demos as a legacy phenomenon, whose goals are now better met by vignettes (@sec-vignettes).
+    For actively maintained packages, it probably makes sense to repurpose the content in any existing demos somewhere that's more visible, e.g. in `README` or in vignettes.
+    These other locations offer other advantages, such as making sure that the code is exercised regularly.
+    This is not true of actual demos, leaving them vulnerable to rot.
 
 -   `exec/`: for executable scripts.
     Unlike files placed in other directories, files in `exec/` are automatically flagged as executable.
+    Empirically, to the extent that R package are shipping scripts for external interpreters, the `inst/` directory seems to be preferred location these days.
 
 -   `po/`: translations for messages.
-    This is useful but beyond the scope of this book.
-    See the [Internationalization](https://cran.rstudio.com/doc/manuals/r-devel/R-exts.html#Internationalization) chapter of "R extensions" for more details.
-
--   `tools/`: auxiliary files needed during configuration, or for sources that need to generate scripts.
+    This is useful, but beyond the scope of this book.
+    See the [Internationalization](https://cran.rstudio.com/doc/manuals/r-devel/R-exts.html#Internationalization) chapter of "Writing R extensions" for more details.
```

Comment:
Worth a link to https://michaelchirico.github.io/potools/index.html too, I think

Reply:
done